src/dftbp/mixer/factory.F90

```diff
@@ -0,0 +1,83 @@
+!--------------------------------------------------------------------------------------------------!
+!  DFTB+: general package for performing fast atomistic simulations                                !
+!  Copyright (C) 2006 - 2023  DFTB+ developers group                                               !
+!                                                                                                  !
+!  See the LICENSE file for terms of usage and distribution.                                       !
+!--------------------------------------------------------------------------------------------------!
+
+#:include 'common.fypp'
+#:set FLAVOURS = [('real', 'real', 'Real'),('cmplx', 'complex', 'Cmplx')]
+module dftbp_mixer_factory
+  use dftbp_io_message, only : error
+  use dftbp_mixer_mixer, only : TMixerReal, TMixerCmplx
+  ! Mixer Input
+  use dftbp_mixer_andersonmixer, only : TAndersonMixerInp
+  use dftbp_mixer_broydenmixer, only : TBroydenMixerInp
+  use dftbp_mixer_diismixer, only : TDiisMixerInp
+  use dftbp_mixer_simplemixer, only : TSimpleMixerInp
+  ! Real Mixer Variants
+  use dftbp_mixer_andersonmixer, only : TAndersonMixerReal, TAndersonMixerReal_init
+  use dftbp_mixer_broydenmixer, only : TBroydenMixerReal, TBroydenMixerReal_init
+  use dftbp_mixer_diismixer, only : TDiisMixerReal, TDiisMixerReal_init
+  use dftbp_mixer_simplemixer, only : TSimpleMixerReal, TSimpleMixerReal_init
+  ! Complex Mixer Variants
+  use dftbp_mixer_andersonmixer, only : TAndersonMixerCmplx, TAndersonMixerCmplx_init
+  use dftbp_mixer_broydenmixer, only : TBroydenMixerCmplx, TBroydenMixerCmplx_init
+  use dftbp_mixer_diismixer, only : TDiisMixerCmplx, TDiisMixerCmplx_init
+  use dftbp_mixer_simplemixer, only : TSimpleMixerCmplx, TSimpleMixerCmplx_init
```

Comment:
Please order imports in alphabetical order by module name (and then also imported items alphabetically...)